packages/experiments/src/components/Announced/examples/Announced.Asynchronous.Example.tsx

```diff
@@ -0,0 +1,177 @@
+import * as React from 'react';
+import { Announced } from '../Announced';
+import { createArray, createRef } from 'office-ui-fabric-react/lib/Utilities';
+import { Image } from 'office-ui-fabric-react/lib/Image';
+import { FocusZone } from 'office-ui-fabric-react/lib/FocusZone';
+import { Toggle } from 'office-ui-fabric-react/lib/Toggle';
+import './Announced.Example.scss';
+import { ProgressIndicator } from 'office-ui-fabric-react/lib/ProgressIndicator';
+
+const DELAY = 10;
+
+export interface IAnnouncedAsynchronousExampleState {
+  photos: { url: string; width: number; height: number }[];
+  total: number;
+  announced?: JSX.Element;
+  percentComplete: number;
+  loading: boolean;
+  complete: boolean;
+  timeSinceLastAnnounce: number;
+}
+
+export interface IAnnouncedAsynchronousExampleProps {}
+
+export class AnnouncedAsynchronousExample extends React.Component<IAnnouncedAsynchronousExampleProps, IAnnouncedAsynchronousExampleState> {
+  private _root = createRef<HTMLElement>();
+  private timer: number;
+  private increaseTotal: number;
+
+  constructor(props: {}) {
+    super(props);
+    this.state = {
+      photos: this._createPhotos(),
+      total: 0,
+      announced: undefined,
+      percentComplete: 0,
+      loading: false,
+      complete: false,
+      timeSinceLastAnnounce: 0
+    };
+
+    this._renderPhotos = this._renderPhotos.bind(this);
+    this._renderAnnounced = this._renderAnnounced.bind(this);
+    this._onFocusPhotoCell = this._onFocusPhotoCell.bind(this);
+    this._onToggleChange = this._onToggleChange.bind(this);
+
+    this.increaseTotal = setInterval(() => {
+      if (this.state.loading && this.state.total < this.state.photos.length) {
+        this.setState({ total: this.state.total + 1 });
+      } else if (this.state.total === this.state.photos.length && this.state.complete !== true) {
+        this.setState({ complete: true });
+        this.setState({ announced: undefined });
```

Comment:
Any reason for not combining these `setState` calls?

Reply:
Nope, no reason! Just an oversight. Good catch.